cmd/thanos/sidecar.go

```diff
@@ -321,20 +305,10 @@ func runSidecar(
 }
 
 func validatePrometheus(ctx context.Context, logger log.Logger, m *promMetadata) error {
-	if m.version == nil {
-		level.Warn(logger).Log("msg", "fetched version is nil or invalid. Unable to know whether Prometheus supports /version endpoint, skip validation")
-		return nil
-	}
-
-	if m.version.LessThan(promclient.FlagsVersion) {
-		level.Warn(logger).Log("msg",
-			"Prometheus doesn't support flags endpoint, skip validation", "version", m.version.Original())
-		return nil
-	}
-
 	flags, err := promclient.ConfiguredFlags(ctx, logger, m.promURL)
```

Comment:
Can we add flow for distincting 404 from that will show that either it's not Prometheus (this will be caught later on, so fine). Maybe via simple sentinel erorr?

Reply:
I am afraid this has to retry ): As we removed first retry which helped us to wait until Prometheus is rdy (often the case). Thanks of first retry you removed we wouldn't need to retry here, as we know Prometheus must be rdy. This is not the case now without retry you removed ):

Reply:
Let me know if my response is clear (:

Reply:
yes. The 404 is a bit fiddly, because we see this. But PTAL